### PR TITLE
Fixes #16924: Reimplement enums

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/16924

### CODE REVIEW

**Comment by peckpeck** [comment]
There will be more than one commit

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit 1 : implement a new enum
Commit 2 : use this new enum
Commit 3: parse a new enum syntax

File: rudder-lang/src/ast/enums.rs [hunk 1]
```diff
@@ -1,283 +1,244 @@
-// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (amousset) [inline rudder-lang/src/ast/enums.rs]: we should keep this

**Comment by peckpeck** [comment]
Ready for review

**Pull request merged**